### PR TITLE
Feat: delete  blog posts

### PR DESCRIPTION
**Pull request by imacpowers**
### What does this PR do?
This PR implements an endpoint for deleting a blog post from the database. The endpoint is restricted to super admin users only and provides appropriate responses for various scenarios.

- **Added DELETE /api/v1/blogs/:id Endpoint**: Allows superadmins to delete existing blog posts.
- **Authentication and Authorization**: Requires a JWT token and superadmin role for access.
- **Response Handling**: Provides appropriate responses for successful deletion, blog not found, unauthorized access, and other error scenarios.

### How should this be manually tested?
**Login Superadmin User**:
1. Make a POST request to `/api/v1/auth/login` with a superadmin user's credentials.
2. Check the response for a valid JWT token.

**Delete Blog Post**:
1. Make a DELETE request to `/api/v1/blogs/:id` with the blog ID in the path parameter and a valid JWT token in the header.
2. Verify the response confirms the blog has been deleted.

**Edge Cases**:
- **Invalid Token**: Test with an invalid token to ensure the endpoint returns a 401 Unauthorized status with the correct message.
- **Missing Token**: Test without a token to ensure the endpoint returns a 401 Unauthorized status with the correct message.
- **Non-superadmin User**: Test with a valid token of a non-superadmin user to ensure the endpoint returns a 403 Forbidden status.
- **Non-existent Blog**: Test with a non-existent blog ID to ensure the endpoint returns a 404 Not Found status.
- **Internal Server Error**: Simulate a server error to ensure the endpoint returns a 500 Internal Server Error status.

### Checklist
- [x] Implemented DELETE `/api/v1/blogs/:id` endpoint in the `blogs.controller.ts`.
- [x] Added `deleteBlogPost` method in the `blogs.service.ts`.
- [x] Modified `blogs.controller.ts` to provide appropriate responses for various cases.
- [x] Tested endpoint functionality to ensure it correctly handles valid and invalid tokens.
- [x] Added appropriate HTTP status codes and response messages for edge cases.
- [x] Added tests for this endpoint.

### Response Examples
**Success**:
```json
{
  "message": "Blog successfully deleted",
  "status_code": 202
}
```

**Forbidden**:
```json
{
  "message": "You are not authorized to perform this action",
  "status_code": 403
}
```

**Not Found**:
```json
{
  "message": "Blog with the given Id does not exist",
  "status_code": 404
}
```

**Internal Server Error**:
```json
{
  "message": "Internal server error.",
  "status_code": 500
}
```

**Invalid Method**:
```json
{
  "message": "This method is not allowed.",
  "status_code": 405
}
```

**Bad Request**:
```json
{
  "message": "An invalid request was sent.",
  "status_code": 400
}
```

### Link to Issue
Link to this issue: #108

### CODE REVIEW

File: src/modules/blogs/blogs.service.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  ) {}
+
+  async createBlog(createBlogDto: CreateBlogDto, user: User): Promise<Blog> {
+    const fullUser = await this.userRepository.findOne({
+      where: { id: user.id },
+      select: ['first_name', 'last_name'],
+    });
+
+    if (!fullUser) {
+      throw new Error('User not found');
```
Comment (Cobidex) [inline src/modules/blogs/blogs.service.ts:24]: please dont throw generic error,  use Custom http exception and return appropriate message and status code, follow the guidlines on the chanel

**Comment by Heba-WebDev** [comment]
Fix the title please
